src/VisualStudio/Core/Def/Implementation/LanguageClient/AlwaysActivateInProcLanguageClient.cs

```diff
@@ -36,26 +38,40 @@ internal class AlwaysActivateInProcLanguageClient : AbstractInProcLanguageClient
             CSharpVisualBasicRequestDispatcherFactory csharpVBRequestDispatcherFactory,
             VisualStudioWorkspace workspace,
             IAsynchronousOperationListenerProvider listenerProvider,
-            ILspWorkspaceRegistrationService lspWorkspaceRegistrationService)
+            ILspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+            DefaultCapabilitiesProvider defaultCapabilitiesProvider)
             : base(csharpVBRequestDispatcherFactory, workspace, diagnosticService: null, listenerProvider, lspWorkspaceRegistrationService, diagnosticsClientName: null)
         {
             _globalOptionService = globalOptionService;
+            _defaultCapabilitiesProvider = defaultCapabilitiesProvider;
         }
 
         public override string Name => "C#/Visual Basic Language Server Client";
 
         protected internal override VSServerCapabilities GetCapabilities()
-            => new VSServerCapabilities
+        {
+            var serverCapabilities = new VSServerCapabilities();
+
+            // If the LSP editor feature flag is enabled advertise support for LSP features here so they are available locally and remote.
+            var isLspEditorEnabled = Workspace.Services.GetRequiredService<IExperimentationService>().IsExperimentEnabled(VisualStudioWorkspaceContextService.LspEditorFeatureFlagName);
+            if (isLspEditorEnabled)
+            {
+                serverCapabilities = _defaultCapabilitiesProvider.GetCapabilities();
+            }
+
+            serverCapabilities.TextDocumentSync = new TextDocumentSyncOptions
             {
-                TextDocumentSync = new TextDocumentSyncOptions
-                {
-                    Change = TextDocumentSyncKind.Incremental,
-                    OpenClose = true,
-                },
-                SupportsDiagnosticRequests = this.Workspace.IsPullDiagnostics(InternalDiagnosticsOptions.NormalDiagnosticMode),
-                // This flag ensures that ctrl+, search locally uses the old editor APIs so that only ctrl+Q search is powered via LSP.
-                DisableGoToWorkspaceSymbols = true,
-                WorkspaceSymbolProvider = true,
+                Change = TextDocumentSyncKind.Incremental,
+                OpenClose = true,
             };
+            serverCapabilities.SupportsDiagnosticRequests = Workspace.IsPullDiagnostics(InternalDiagnosticsOptions.NormalDiagnosticMode);
+
+            // When using the lsp editor, set this to false to allow LSP to power goto.
+            // Otherwise set to true to disable LSP for goto
+            serverCapabilities.DisableGoToWorkspaceSymbols = !isLspEditorEnabled;
+            serverCapabilities.WorkspaceSymbolProvider = true;
```

Comment:
why is this uniformly set to true, instead of using the 'enabled' flag.  What is the difference between these two values?  i don't get how we're enabling workspacesymblprovider, but then adding a flag to possibly disable it.  super weird.

Reply:
'WorkspaceSymbolProvider' is uniformly true to allow LSP to power cntrl+Q search, will add a comment there.  

Normally, setting 'WorkspaceSymbolProvider' to true would mean that LSP also provides go to cntrl+, search results.  So the platform added 'DisableGoToWorkspaceSymbols ' to allow servers to prevent LSP from handling cntrl+, while using LSP for cntrl+Q.  So DisableGoToWorkspaceSymbols  needs to check the feature flag (if on, use LSP for cntrl+, as well)

Reply:
Please comment. That's not at all evident